src/plugins/formulas/autofill.js

```diff
@@ -4,151 +4,152 @@
  * @param {object} pluginInstance The formulas plugin instance.
  */
 export const registerAutofillHooks = (pluginInstance) => {
-  const lastAutofillSource = { value: undefined };
+  /**
+   * The array of arrays used to check if no values were returned from
+   * `beforeAutofill`, other than our own.
+   * */
+  const sentinel = [[]];
 
   // Block autofill operation if at least one of the underlying's cell
   // contents cannot be set, e.g. if there's a matrix underneath.
-  pluginInstance.addHook('beforeAutofill', (start, end) => {
-    const width = Math.abs(start.col - end.col) + 1;
-    const height = Math.abs(start.row - end.row) + 1;
+  pluginInstance.addHook('beforeAutofill', (_, __, target) => {
+    const width = target.to.col - target.from.col + 1;
+    const height = target.to.row - target.from.row + 1;
 
-    const row = Math.min(start.row, end.row);
-    const col = Math.min(start.col, end.col);
+    const row = target.from.row;
+    const col = target.from.col;
 
     if (
-      !pluginInstance.engine.isItPossibleToSetCellContents({
+      pluginInstance.engine.isItPossibleToSetCellContents({
```

Comment:
That's because your later hook overwrote the `false` result. 

```ts
hot.addHook('beforeAutofill', (arg) => false); // 1
hot.addHook('beforeAutofill', (arg) => true); // 2
hot.addHook('beforeAutofill', (arg) => { console.log('third hook') }); // 3
```

Our internal plugin is registered first so it's (1) and as an arg gets `selectedData`. User register second hook (2) and gets an `arg === false` but can overwrite it with `true`. Third hook return nothing, the last returned arg will be used. That's fine.

Here we will return the `sentinel` when ok, but `undefined` otherwise. So if the autofill is not possible in HyperFormula we will not apply data in the `afterAutofill` hook. But if user has registered other `beforeAutofill` he cannot know that the action will not happen. Our hook should return `false` that will be propagated through all the hooks in the bucket and the developer can cancel his changes.

Reply:
> Our hook should return false that will be propagated through all the hooks in the bucket and the developer can cancel his changes

@wojciechczerniak How can they check for that?

Reply:
The first `arg` is `false`

Reply:
I see, thanks for the explanations, https://github.com/handsontable/handsontable/pull/7987/commits/09cb6b7cf2f515e838291a76cff079aeb9dc26b2